templates/default/deploy-scripts/src/main.rs

```diff
@@ -0,0 +1,126 @@
+pub mod deployer;
+use anyhow::{Context, Error};
+use clap::Parser;
+use concordium_rust_sdk::{
+    common::types::Amount,
+    smart_contracts::{
+        common::{self as contracts_common},
+        types::{OwnedContractName, OwnedParameter, OwnedReceiveName},
+    },
+    types::{
+        smart_contracts::{ModuleReference, WasmModule},
+        transactions,
+        transactions::send::GivenEnergy,
+    },
+    v2,
+};
+use itertools::Itertools;
+
+use concordium_rust_sdk::types::transactions::InitContractPayload;
+use deployer::Deployer;
+use std::{
+    io::Cursor,
+    path::{Path, PathBuf},
+};
+
+/// Reads the wasm module from a given file path and file name.
+fn get_wasm_module(file: &Path) -> Result<WasmModule, Error> {
+    let wasm_module = std::fs::read(file).context("Could not read the WASM file")?;
+    let mut cursor = Cursor::new(wasm_module);
+    let wasm_module: WasmModule = concordium_rust_sdk::common::from_bytes(&mut cursor)?;
+    Ok(wasm_module)
+}
+
+/// Command line flags.
+#[derive(clap::Parser, Debug)]
+#[clap(author, version, about)]
+struct App {
+    #[clap(
+        long = "node",
+        default_value = "http://node.testnet.concordium.com:20000",
+        help = "V2 API of the Concordium node."
+    )]
+    url: v2::Endpoint,
+    #[clap(
+        long = "account",
+        help = "Path to the file containing the Concordium account keys exported from the wallet \
+                (e.g. ./myPath/3PXwJYYPf6fyVb4GJquxSZU8puxrHfzc4XogdMVot8MUQK53tW.export)."
+    )]
+    key_file: PathBuf,
+    #[clap(
+        long = "module",
+        help = "Path of the Concordium smart contract module. Use this flag several times if you \
+                have several smart contract modules to be deployed (e.g. --module \
+                ./myPath/default.wasm.v1 --module ./default2.wasm.v1)."
+    )]
+    module: Vec<PathBuf>,
+}
+
+/// Main function: It deploys to chain all wasm modules from the command line
+/// `--module` flags. Write your own custom deployment/initialization script in
+/// this function. An deployment/initialization script example is given in this
+/// function for the `default` smart contract.
+#[tokio::main]
+async fn main() -> Result<(), Error> {
+    let app: App = App::parse();
+
+    let concordium_client = v2::Client::new(app.url).await?;
+
+    let mut deployer = Deployer::new(concordium_client, &app.key_file)?;
+
+    let mut modules_deployed: Vec<ModuleReference> = Vec::new();
+
+    for contract in app.module.iter().unique() {
```

Comment:
Is there a reason why we do `unique` here as opposed to doing what the user requested?

Seems to me like an odd place to filter this out, especially since it's filtering out by file name, rather than by module hash.

I think it would be better if the deploy_wasm_module function simply checked if what you're deploying already exists, and that's it. We don't need to complicate it here by filtering out duplicate file names.

Reply:
removed. 

The idea was to prevent people from accidentally deploying the same module twice but we do some checks if a module already exists on chain as well, so it seems like not needed this check now.